src/libnet_if_addr.c

```diff
@@ -46,7 +46,9 @@
 
 #include "../include/ifaddrlist.h"
 
-#define MAX_IPADDR 512
+#define MAX_IP_ADDR 512
+#define IPADDR_INITIAL_BUFSIZE MAX_IP_ADDR
+#define IPADDR_GROW_FACTOR 1.5
 
```

Comment:
Please change this to not use floating point.  I'd rather see `IP_ADDR_GROW 256` or something.

To track number of current addresses, use a global (to this file) construct:

```c
static size_t ip_addr_num = MAX_IP_ADDR;
```

Reply:
The idea of the floating point growth factor is to scale better to a large number of interfaces.

Say we have 3000 interfaces/IP addresses:

With static 256 growth factor:
`(3000 - 512)/256 = 9.7`
we will need already 10 realloc

with the multiply 1.5 growth factor:
`512 * (1.5)ˆx >= 3000`
`x >= 4.5`
Only need 5 reallocs

1.5 was chosen as a good ratio for dynamic array growth and previous left memory holes reuse (https://stackoverflow.com/questions/1100311/what-is-the-ideal-growth-rate-for-a-dynamically-allocated-array)

Let me know what you think and if you'd rather keep the constant addition. I will definitely make the global `ip_addr_num` change in both cases.

Thank you very much for the review

Reply:
OK that's a good motivation, I appreciate your explanation.  I basically just dislike the use of floating point, so if you can address that I'm happy. Eg num += num >> 2; or something.

Reply:
That's a good idea, thank you! I also felt an umconfortable shiver in my body when I started using floating points. Never know when NaN or 0.000001 is going to creep up on you.